absl/strings/str_cat.cc

```diff
@@ -89,7 +89,7 @@ static char* Append(char* out, const AlphaNum& x) {
   // memcpy is allowed to overwrite arbitrary memory, so doing this after the
   // call would force an extra fetch of x.size().
   char* after = out + x.size();
-  memcpy(out, x.data(), x.size());
+  if (x.size()) memcpy(out, x.data(), x.size());
```

Comment:
Please use !x.empty(), and please use curly braces around the if statement.

Reply:
@derekmauro  Fixed style comments.  please note AlphaNum does not have `.empty()`.